pysph/examples/rigid_body/sphere_in_vessel_akinci.py

```diff
@@ -0,0 +1,220 @@
+
+    xr = np.arange(100, 100 + 2 * dx, dx)
+    yr = np.arange(0, 250, dx)
+    xr, yr = np.meshgrid(xr, yr)
+    xr = xr.ravel()
+    yr = yr.ravel()
+
+    x = np.concatenate([xl, xb, xr])
+    y = np.concatenate([yl, yb, yr])
+
+    return x * 1e-3, y * 1e-3
+
+
+def create_fluid():
+    dx = 2
+    xf = np.arange(0, 100, dx)
+    yf = np.arange(0, 150, dx)
+    xf, yf = np.meshgrid(xf, yf)
+    xf = xf.ravel()
+    yf = yf.ravel()
+
+    return xf * 1e-3, yf * 1e-3
+
+
+def create_sphere(dx=1):
+    x = np.arange(0, 100, dx)
+    y = np.arange(151, 251, dx)
+    x, y = np.meshgrid(x, y)
+    x = x.ravel()
+    y = y.ravel()
+
+    p = ((x - 50)**2 + (y - 200)**2) < 20**2
+    x = x[p]
+    y = y[p]
+
+    # lower sphere a little
+    y = y - 20
+    return x * 1e-3, y * 1e-3
+
+
+def get_density(y):
+    height = 150
+    c_0 = 2 * np.sqrt(2 * 9.81 * height * 1e-3)
+    rho_0 = 1000
+    height_water_clmn = height * 1e-3
+    gamma = 7.
+    _tmp = gamma / (rho_0 * c_0**2)
+
+    rho = np.zeros_like(y)
+    for i in range(len(rho)):
+        p_i = rho_0 * 9.81 * (height_water_clmn - y[i])
+        rho[i] = rho_0 * (1 + p_i * _tmp)**(1. / gamma)
+    return rho
+
+
+def geometry():
+    # please run this function to know how
+    # geometry looks like
+    x_tank, y_tank = create_boundary()
+    x_fluid, y_fluid = create_fluid()
+    x_cube, y_cube = create_sphere()
+    plt.scatter(x_fluid, y_fluid)
+    plt.scatter(x_tank, y_tank)
+    plt.scatter(x_cube, y_cube)
+    plt.axes().set_aspect('equal', 'datalim')
+    print("done")
+    plt.show()
+
+
+class RigidFluidCoupling(Application):
+    def initialize(self):
+        self.dx = 2 * 1e-3
+        self.hdx = 1.2
+        self.ro = 1000
+        self.solid_rho = 100
+        self.m = 1000 * self.dx * self.dx
+        self.co = 2 * np.sqrt(2 * 9.81 * 150 * 1e-3)
+        self.alpha = 0.1
+
+    def create_particles(self):
+        """Create the circular patch of fluid."""
+        xf, yf = create_fluid()
+        rho = get_density(yf)
+        m = rho[:] * self.dx * self.dx
+        rho = np.ones_like(xf) * self.ro
+        h = np.ones_like(xf) * self.hdx * self.dx
+        fluid = get_particle_array_wcsph(x=xf, y=yf, h=h, m=m, rho=rho,
+                                         name="fluid")
+
+        dx = 2
+        xt, yt = create_boundary()
+        m = np.ones_like(xt) * 1000 * self.dx * self.dx
+        rho = np.ones_like(xt) * 1000
+        rad_s = np.ones_like(xt) * 2 / 2. * 1e-3
+        h = np.ones_like(xt) * self.hdx * self.dx
+        V = np.ones_like(xt) * dx * dx * 1e-6
+        tank = get_particle_array_wcsph(x=xt, y=yt, h=h, m=m, rho=rho,
+                                        rad_s=rad_s, V=V, name="tank")
+        for name in ['fx', 'fy', 'fz']:
+            tank.add_property(name)
+
+        dx = 1
+        xc, yc = create_sphere(1)
+        m = np.ones_like(xc) * self.solid_rho * dx * 1e-3 * dx * 1e-3
+        rho = np.ones_like(xc) * self.solid_rho
```

Comment:
With latest pysph you do not need ones_like for simple constants, just pass in a constant for the constant properties and it should work.  It will broadcast this to the same shape as x, y, z and use it.